File: src/main/java/mediathek/gui/duplicates/CommonStatsEvaluationTask.java
package mediathek.gui.duplicates;

import ca.odell.glazedlists.TransactionList;
import mediathek.config.Daten;
import mediathek.daten.DatenFilm;

import java.util.Map;
import java.util.stream.Collectors;

public class CommonStatsEvaluationTask implements Runnable {
    @Override
    public void run() {
        final var films = Daten.getInstance().getListeFilme().parallelStream()
                .filter(f -> !f.isLivestream())
                .toList();

        var statisticsList = Daten.getInstance().getCommonStatistics();

        Map<String, Long> statisticsMap = films.parallelStream()
                .collect(Collectors.groupingBy(DatenFilm::getSender, Collectors.counting()));
        TransactionList<FilmStatistics> tList = new TransactionList<>(statisticsList);
        statisticsList.getReadWriteLock().writeLock().lock();
        tList.beginEvent(true);
        tList.clear();
        for (var sender : statisticsMap.keySet()) {
            tList.add(new FilmStatistics(sender, statisticsMap.get(sender)));
        }
        tList.commitEvent();
        statisticsList.getReadWriteLock().writeLock().unlock();
    }
}

File: src/main/kotlin/mediathek/gui/duplicates/CommonStatsEvaluationTask.kt
package mediathek.gui.duplicates

import mediathek.filmlisten.FilmCatalog

class CommonStatsEvaluationTask(
    private val filmCatalog: FilmCatalog,
) : Runnable {
    override fun run() {
        val statisticsMap = filmCatalog.allFilms.parallelStream()
            .filter { film -> !film.isLivestream }
            .countFilmsBySender()

        replaceFilmStatistics(filmCatalog.commonStatistics, statisticsMap)
    }
}
